job_executor/domain/rollback.py

```diff
@@ -177,17 +177,23 @@ def rollback_worker_phase_import_job(
 def rollback_manager_phase_import_job(
     job_id: str, operation: str, dataset_name: str
 ):
-    logger.info(
-        f'{job_id}: Rolling back import job '
-        f'with target: "{dataset_name}" and operation "{operation}"'
-    )
-    logger.info(f'{job_id}: Restoring files from temporary backup')
-    local_storage.restore_from_temporary_backup()
+    try:
+        logger.info(
+            f'{job_id}: Rolling back import job '
+            f'with target: "{dataset_name}" and operation "{operation}"'
+        )
+        logger.info(f'{job_id}: Restoring files from temporary backup')
+        local_storage.restore_from_temporary_backup()
 
-    logger.info(f'{job_id}: Deleting metadata draft file')
-    local_storage.delete_metadata_draft(dataset_name)
-    if operation in ['ADD', 'CHANGE']:
-        logger.info(f'{job_id}: Deleting data file/directory')
-        local_storage.delete_parquet_draft(dataset_name)
-    logger.info(f'{job_id}: Deleting temporary backup')
-    local_storage.delete_temporary_backup()
+        logger.info(f'{job_id}: Deleting metadata draft file')
+        local_storage.delete_metadata_draft(dataset_name)
+        if operation in ['ADD', 'CHANGE']:
+            logger.info(f'{job_id}: Deleting data file/directory')
+            local_storage.delete_parquet_draft(dataset_name)
+        logger.info(f'{job_id}: Deleting temporary backup')
+        local_storage.delete_temporary_backup()
+    except Exception as e:
```

Comment:
I think this should actually fail catastrophically if it fails. When we do the startup checking and rollback:
1. App starts
2. Tries to roll back interrupted jobs
3. uses this function and it fails, but exception is handled
4. Starts app normally

If this fails it should signal to the code using it that it failed, (and in that case it is probably OK to crash the app:
3.  uses this function and it fails
4. app won't start cause rollback fails

Similarily if a job fails and needs to roll back and the rollback fails it should propagate all the way back to the root app.py and shut down the app. Due to the catastrophic nature of a rollback failing.

Reply:
True, if the rollback fails, we don't want automation here, we want to manually check what has happened. I'm putting a comment in the code on this.